src/screens/Menu/Settings/Settings.js

```diff
@@ -0,0 +1,147 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2021 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import * as React from 'react';
+import { useNavigation } from 'react-navigation-hooks';
+import { useDispatch } from 'react-redux';
+import styled from 'styled-components/native';
+import { useTranslationWithPrefix } from 'translations/translate';
+
+// Components
+import { Container, Content, Spacing } from 'components/modern/Layout';
+import CheckAuth from 'components/CheckAuth';
+import HeaderBlock from 'components/HeaderBlock';
+import Text from 'components/modern/Text';
+
+// Selectors
+import { useRootSelector } from 'selectors';
+
+// Actions
+import { resetIncorrectPasswordAction } from 'actions/authActions';
+
+// Utils
+import { getKeychainDataObject } from 'utils/keychain';
+import { appFont, fontStyles, spacing } from 'utils/variables';
+
+// Types
+import type { WalletObject } from 'models/Wallet';
+
+// Local
+import DarkModeSetting from './components/DarkModeSetting';
+import LanguageSetting from './components/LanguageSetting';
+import CurrencySetting from './components/CurrencySetting';
+import BiometricLoginSetting from './components/BiometricLoginSetting';
+import ChangePinSetting from './components/ChangePinSetting';
+import ViewBackupPhraseSetting from './components/ViewBackupPhraseSetting';
+import BackupWalletSetting from './components/BackupWalletSetting';
+
+const Settings = () => {
+  const { t } = useTranslationWithPrefix('menu.settings');
+  const navigation = useNavigation();
+
+  const { showPinModal, pin, wallet, onPinValid } = useWalletData();
+
+  if (showPinModal) {
```

Comment:
My concern here is that we request PIN for each Settings entry even if user wants to change just language, we can take this discussion to Slack.